### ISSUE TITLE
Additions to the French Messagease keyboard

### ISSUE THREAD

**Issue by gynl** [dessalines/thumb-key#675]
### Describe your suggested feature

Hi,

First of all, thank you for your work. It’s really great to see this app with a public licence !

I just make some adds in French Messagease layout :
– add `«` char on `t` and `T` bottom left (under `<` ) ;
– add `»` on `s` and `S` bottom right (under `>`) ;
– add NNBSP on `i` and `I`  right — used for long string of numbers or before hight punctuation mark (?!; &c.) in french ;
– add `’` wich is french version of the `'` used in english on `i` and `I` top right ;
– add `—` which is almost used the same way as parenthesis on `a` and `A` left (near `-`) ;

I think these chars are usefull in french and need to be added. Could you please update FR Messagease layout with this one ; or add this as a new version with the next release ?

[FRMessageEase.kt.txt](https://github.com/dessalines/thumb-key/files/14177115/FRMessageEase.kt.txt)

Thank you !

### Acknowledgements

- [X] I have searched the existing issues and this is a new one, **NOT** a duplicate or related to another open issue.
- [X] This is a **single** feature request,  in case of multiple feature requests I will open a separate issue for each one (they can always link to each other if related)
- [X] This is not a question or a discussion, in which case I should have gone to [lemmy.ml/c/thumbkey](https://lemmy.ml/c/thumbkey)
- [X] I have written a short but informative title.
- [X] I have checked through the app settings for my feature.
- [ ] I have admitted that I am a clown by having checked this box, as I have not read these acknowledgements. 🤡
- [X] I will fill out all of the requested information in this form.

**Comment by breversa** [comment]
Hi!
As the original author of the french layout version, I had suggested numerous improvements/extra characters to ease up the speed of typing, but some were refused, arguing that they would be better placed in the special characters’ screen.

Your suggestions, however, are specific to french (I’m pretty sure you’re a bépo user, aren’t you? ;-) ) so they would definitely fit in the letters screen… but have you checked the current FR layout?

- The `«` and `»` characters are already here (top-left square).
- How would you visually represent NNBSP on the square?
- I’m not sure if the character in the top-left square is `'` or `’` but I do remember it was mentioned during development.
- Not sure about `—`. I probably wouldn’t use myself, but I wouldn’t object unless it clutters the screen too much, or takes the place of a another, more useful character.

**Comment by gynl** [comment]
Hi breversa, Happy to talk with you.

Bépo user... hmm... I’m busted! :-D
I'll be honest: coming from Messagease, I haven't bothered to study other French configurations yet. I'll probably do it at some point, but right now, I don't have the energy to learn and master a new one (I know, I should).

For NNBSP, I've used `␣` (u2423); I admit it might not be the best choice, but I don't really see any other options either.

For `—`, I agree. I personally use it a lot, but I'm probably a rare case. I took a free spot thinking... why not?

Interestingly, the diaeresis (`¨`) is also absent from the French version of FR Messagease... I didn't notice it right away.

Anyway, I think this fork must have a layout editor… Sure I’m not the only one thinking that and it will come one day. I'm not able to do it to contribute… so…

**Comment by renyhp** [comment]
> Anyway, I think this fork must have a layout editor… Sure I’m not the only one thinking that and it will come one day. I'm not able to do it to contribute… so…

@gynl I agree with you there, and even if it is quite obvious you're not the only one thinking this, it looks like the dev disagrees. See discussion in #657, https://github.com/dessalines/thumb-key/issues/645 , https://github.com/dessalines/thumb-key/issues/586 , https://github.com/dessalines/thumb-key/issues/351 , https://github.com/dessalines/thumb-key/issues/290 , https://github.com/dessalines/thumb-key/issues/160 ...

In other words it really looks like the dev prefers to deal themself with users proposing new layouts, adding new layouts, proposing changes to existing layouts, and having the repo filled with issues and PRs about layouts, rather than giving this freedom to the users (he raises technical issues which frankly I just think they are not problems at all). So if users want to have a user-friendly way to edit their keyboards, they will have to wait for [open-thumb](https://github.com/asdkant/open-thumb) to be developed...

**Comment by kke** [comment]
> So if users want to have a user-friendly way to edit their keyboards, they will have to wait for [open-thumb](https://github.com/asdkant/open-thumb) to be developed.

Or maybe someone could create a web based editor that would output `.kt` files to make it easier to submit PRs 😀

**Comment by luc--** [comment]
It would also be nice to add œ to fr-messagease. Most annoying part with the absence of the (¨).

**Comment by florencebiree** [comment]
The french layouts  also miss the "·" character (medium point?). I use it a lot, so it's a big problem for me to use the keyboard...

**Comment by luc--** [comment]
> The french layouts also miss the "·" character (medium point?). I use it a lot, so it's a big problem for me to use the keyboard...

It should rather be in the greek layout (where it's missing if I'm not wrong) as it's the equivalent of ";" for greek. Not commonly used in French except for epigraphy probably.

**Comment by luc--** [comment]
> In French, it is a lot used for inclusive language for people who use it (not all french people, but as a person who use it, I use it a lot). Example: trois gentil·les boulanger·es. This system, even if non official, is used by an increasing amount of people over time (some newspaper use it, as example).
> 
> It is easily accessible in the bépo layout I use on my computer, and I love it, and also in an AZERTY layout on AnySoftKeyboard on Android (wich I use, even if I would prefer to switch to Thumb Key). Somewhere near the lower dot "." is IMHO a good place, not far away in special symbols.

Ok if it's not conventional French, it should only be left as an option then as it is not commonly used, in order to avoid unwanted typos. Btw I don't think it was supported in Messagease original keyboard.

**Comment by dessalines** [comment]
Can re-open if someone wants to make these changes.
